CentralUnit/Communication/XbeeNetwork.cs

```diff
@@ -0,0 +1,416 @@
+        /// </summary>
+        /// <param name="packet">The <see cref="XBeePacket"/> to process.</param>
+        private void ProcessXbeePacket(XBeePacket packet)
+        {
+            if (packet != null)
+            {
+                if (packet.GetType() == typeof(ExplicitRxIndicatorPacket))
+                {
+                    ExplicitRxIndicatorPacket eriPacket = (ExplicitRxIndicatorPacket)packet;
+                    XbeeSerialCommunication deviceForPacket = this.GetDevice(eriPacket.SourceAddress64);
+                    deviceForPacket.OnDataReceived(eriPacket.RFData);
+                }
+                else if (packet.GetType() == typeof(TransmitStatusPacket))
+                {
+                    TransmitStatusPacket txsPacket = (TransmitStatusPacket)packet;
+                    this.ProcessTransmitStatusPacket(txsPacket);
+                }
+                else
+                {
+                    Log.InfoFormat("Got unknown packet of type {0}", packet.GetType().Name);
+                    if (Log.IsDebugEnabled)
+                    {
+                        Log.Debug(packet.ToPrettyString());
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Process a received <see cref="TransmitStatusPacket"/>.
+        /// On success, only debug logging occurs.
+        /// When transmission is indicated as not successful, a failure is reported for the device that sent the frame identified in the <see cref="TransmitStatusPacket"/>
+        /// </summary>
+        /// <param name="txsPacket">The <see cref="TransmitStatusPacket"/> parsed from the byte array.</param>
+        private void ProcessTransmitStatusPacket(TransmitStatusPacket txsPacket)
+        {
+            if (this.devices.Any(dev => dev.FrameID == txsPacket.FrameID))
+            {
+                XbeeSerialCommunication deviceForPacket = this.devices.First(dev => dev.FrameID == txsPacket.FrameID);
+
+                if (txsPacket.TransmitStatus == XBeeLibrary.Core.Models.XBeeTransmitStatus.SUCCESS)
+                {
+                    if (Log.IsDebugEnabled)
+                    {
+                        Log.DebugFormat("Transmit succeeded to {0}", deviceForPacket.Xbee64address.ToString());
+                    }
+                }
+                else
+                {
+                    Log.WarnFormat("Got TX status {0} for device {1}", txsPacket.TransmitStatus, deviceForPacket.Xbee64address.ToString());
+                    deviceForPacket.OnFailure();
+                }
+            }
+            else
+            {
+                Log.WarnFormat("Got TX Status {0} for unknown frame ID {1}", txsPacket.TransmitStatus, txsPacket.FrameID);
+            }
+        }
+
+        /// <summary>
+        /// Parse packets in the receive queue and distribute them among the devices in the network.
+        /// </summary>
+        private void ManageRxQueue()
```

Comment:
Are RX and TX standard terms in serial communication?

Reply:
Yes. Rx = Receive, Tx = Transmit